### PR TITLE
minor error in validation of slot and ep fixed

### PR DESCRIPTION
**Pull request by fysnet**

### CODE REVIEW

**Comment by fysnet** [comment]
It looks like lines 1644, 3063, and 3067 had non-standard single-quote characters around the numbers '1' and '0'. I have since fixed this in my code so that it won't show up again.

File: bochs/iodev/usb/usb_xhci.cc [hunk 1]
```diff
@@ -1641,7 +1641,7 @@ bool bx_usb_xhci_c::write_handler(bx_phy_address addr, unsigned len, void *data,
             BX_XHCI_THIS hub.usb_port[port].portsc.wpr = (value & (1 << 31)) ? 1 : 0;
             BX_XHCI_THIS hub.usb_port[port].portsc.cec = (value & (1 << 23)) ? 1 : 0;
             BX_XHCI_THIS hub.usb_port[port].portsc.wrc = (value & (1 << 19)) ? 0 : BX_XHCI_THIS hub.usb_port[port].portsc.wrc;
-            if (value & (1<<18))  // For a USB3 protocol port, this bit shall never be set to �1�, however doesn't hurt to write to it anyway
+            if (value & (1<<18))  // For a USB3 protocol port, this bit shall never be set to 1, however doesn't hurt to write to it anyway
```
Comment (stlintel) [inline bochs/iodev/usb/usb_xhci.cc:1644]: these are unprintable characters, what did you wan to use here? '1' ?

File: bochs/iodev/usb/usb_xhci.cc [hunk 2]
```diff
@@ -3060,11 +3060,11 @@ int bx_usb_xhci_c::validate_ep_context(const struct EP_CONTEXT *ep_context, cons
         if (ep_context->tr_dequeue_pointer == 0)
           ret = PARAMETER_ERROR;
         
-        // 4) the DCS field = �1�,
+        // 4) the DCS field = 1,
```
Comment (stlintel) [inline bochs/iodev/usb/usb_xhci.cc:3063]: unprintable characters

File: bochs/iodev/usb/usb_xhci.cc [hunk 3]
```diff
@@ -3060,11 +3060,11 @@ int bx_usb_xhci_c::validate_ep_context(const struct EP_CONTEXT *ep_context, cons
...
         if (ep_context->dcs != 1)
           ret = PARAMETER_ERROR;
         
-        // 5) the MaxPStreams field = �0�, and
+        // 5) the MaxPStreams field = 0, and
```
Comment (stlintel) [inline bochs/iodev/usb/usb_xhci.cc:3067]: unprintable characters

**Review by stlintel — CHANGES_REQUESTED** [review]

**Comment by stlintel** [comment]
> It looks like lines 1644, 3063, and 3067 had non-standard single-quote characters around the numbers '1' and '0'. I have since fixed this in my code so that it won't show up again.

They still looks non-ASCII characters on github review page(

**Comment by fysnet** [comment]
I thought I had changed them to a single quote, a standard single quote, but apparently I did not. Now I have simply removed the char.

File: bochs/iodev/usb/usb_xhci.cc (hunk 2 above)
Comment (fysnet) [reply bochs/iodev/usb/usb_xhci.cc]: removed the unprintable char

File: bochs/iodev/usb/usb_xhci.cc (hunk 3 above)
Comment (fysnet) [reply bochs/iodev/usb/usb_xhci.cc]: removed the unprintable char

File: bochs/iodev/usb/usb_xhci.cc (hunk 1 above)
Comment (fysnet) [reply bochs/iodev/usb/usb_xhci.cc]: removed the unprintable char

**Review by stlintel — APPROVED** [review]

**Pull request merged**